phoenix-core/src/it/java/org/apache/phoenix/jdbc/LoggingConnectionLimiterIT.java

```diff
@@ -186,6 +187,55 @@ public void testActivityLogsOnQueryWhenFailures() throws Exception {
 
   }
 
+  @Test
+  public void testQueryExplainPlan() throws Exception {
+
+    String query = "SELECT * FROM " + tableName; // FULL SCAN
+
+    if (getConnection() instanceof PhoenixConnection) {
+      try (PhoenixConnection pconn = getConnection().unwrap(PhoenixConnection.class);
+        Statement stmt = pconn.createStatement()) {
+        ResultSet rs = stmt.executeQuery(query);
+        while (rs.next()) {
+          // do nothing
+        }
+        boolean queryPlanFound = false;
+        String queryPlan = pconn.getActivityLogger().getExplainPlanInfo();
```

Comment:
Can we add a test in some other IT class where we can validate that queryPlan is null by default?

Reply:
@virajjasani added the UT